app/controllers/posts_controller.rb

```diff
@@ -35,6 +35,14 @@ def create
     end
   end
 
+  def destroy
+    current_user = User.find(params[:author_id])
+    @post = Post.find(params[:id])
+    authorize! :destroy, @post
```

Comment:
- [ OPTIONAL ] Great job Team. You flawlessly handled authorization in your posts controller great job :+1: I just wanted to suggest another approach you could read on which is the `load_and_authorize_resource` method from `CanCanCan`.  Using `load_and_authorize_resource` will automatically handle the `loading` and `authorization` of the resource (in this case, the post) based on the controller actions and the user's abilities defined in the ability file. :smile: 
- An example Implementation
```ruby
class PostsController < ApplicationController
  load_and_authorize_resource

  # The rest of your controller actions...
end
```